platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+#############################################################################
+# Mellanox
+#
+# Module contains an implementation of SONiC Platform Base API and
+# provides the Chassis information which are available in the platform
+#
+#############################################################################
+
+import sys
+
+try:
+    from sonic_platform_base.chassis_base import ChassisBase
+    from sonic_platform.psu import Psu
+except ImportError as e:
+    raise ImportError (str(e) + "- required module not found")
+
+MLNX_NUM_PSU = 2
+
+class Chassis(ChassisBase):
+    """Platform-specific Chassis class"""
+    def __init__(self):
+        ChassisBase.__init__(self)
+        self._psu_list.append(None)
+        for index in range(1, MLNX_NUM_PSU + 1):
```

Comment:
Ok, just not aware of the discussion you had for this change. I think 0-based makes more sense. :) But we may need to update the [design doc](https://github.com/Azure/SONiC/wiki/The-PSU-CLI-and-SNMP-support) to notice the vendors about this.

Reply:
Agreed. I think it makes sense to count all devices with base 0 in this API.

If we think it's more "user-friendly" to display information 1-based to the end user, then we should translate from 0-based to 1-based in the appropriate `show` commands.

Reply:
Is SNMP going to use this same API ?   If it is, we might need to consider that SNMP is 1 based for table index.

Reply:
@JohnFanDNI: Yes, SNMP will use this API, much like it currently uses the existing platform API. I believe that was the main reason for using a 1-based index for PSUs in the first place. However, I think it will be better to unify the API and use 0-based indexing for all devices. The SNMP sub-agent will simply need to be modified to add 1 to all PSU indices. Do you see an issue with this? The alternative would be to have to platform API reference all devices using 1-based indexing, which I feel would needlessly complicate the API.